### ISSUE TITLE
Allow self-closing slot element

### ISSUE THREAD

**Issue by hirokiosame** [vuejs/eslint-plugin-vue#151]
**Tell us about your environment**

* **ESLint Version:** 3.19.0
* **eslint-plugin-vue Version:** 3.11.0
* **Node Version:** 7.9.0

**Please show your full configuration:**
```
'vue/html-self-closing': ['error'],
```
**What did you do? Please include the actual source code causing the issue.**

```vue
<template>
	<div>
		<slot />
	</div>
</template>
```

**What did you expect to happen?**
For it to allow self-closing Vue tags like `<slot>`.

**What actually happened? Please include the actual, raw output from ESLint.**
` error  Disallow self-closing on HTML elements (<slot/>)  vue/html-self-closing`

**Comment by mysticatea** [comment]
Thank you for this issue.

`<slot>` is one of HTML elements: https://html.spec.whatwg.org/multipage/scripting.html#the-slot-element
So you can specify it by `html.normal` option.

```js
{
    "html-self-closing": ["error", {
        "html": {
            "normal": "always"
        }
    }]
}
```

**Comment by hirokiosame** [comment]
Interesting. I wanted to argue that Vue is a custom "Web Component" implementation and thus overriding and implementing it's own version of `<slot>` but I guess it follows the spec quite closely. I think the clearest distinction that `<slot>` is not really the HTML version of `<slot>` is the scoped-slot feature:
```
  <slot name="item"
    v-for="item in items"
    :text="item.text">
    <!-- fallback content here -->
  </slot>
```
